### PR TITLE
feat: add support to handle blocking alerts in the alert modal

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

This PR aims to add support to handle blocking alerts to the `AlertModal` component.
Changes:
- added a property called `isBlocking`
- added logic to hide the `Got it` button and acknowledgement checkbox for blocking alerts

**Context**
Conceptually, some alerts cannot be acknowledged or skipped since they indicate the relevant process cannot continue. For example, having an insufficient balance to cover the gas fees for a transaction.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24400?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2411

## **Manual testing steps**

The alert modal is currently not in use and is not available to users. It is part of the new alert system, and we will have a dedicated PR to integrate this component into the new confirmation screens.

## **Screenshots/Recordings**

[blocking-alert.webm](https://github.com/MetaMask/metamask-extension/assets/45455812/a7587883-a542-487c-bc99-1ada575943b2)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]

File: ui/ducks/confirm-alerts/confirm-alerts.ts [hunk 1]
```diff
@@ -14,6 +14,12 @@ export type Alert = {
    */
   alertDetails?: string[];
 
+  /**
+   * Whether the alert is a blocker and un-acknowledgeable, preventing
+   * the user from proceeding and relying on the actions to proceed.
```
Comment (digiwand) [inline ui/ducks/confirm-alerts/confirm-alerts.ts:19]:
```suggestion
   * the user from proceeding and relying on actions to proceed.
```

File: ui/ducks/confirm-alerts/confirm-alerts.ts [hunk 2]
```diff
@@ -14,6 +14,12 @@ export type Alert = {
    */
   alertDetails?: string[];
 
+  /**
+   * Whether the alert is a blocker and un-acknowledgeable, preventing
+   * the user from proceeding and relying on actions to proceed.
+   */
+  isBlocking?: boolean;
```
Comment (matthewwalsh0) [inline ui/ducks/confirm-alerts/confirm-alerts.ts:21]: Minor, but alphabetical order?

File: ui/ducks/confirm-alerts/confirm-alerts.ts [hunk 3]
```diff
@@ -14,6 +14,12 @@ export type Alert = {
    */
   alertDetails?: string[];
 
+  /**
+   * Whether the alert is a blocker and un-acknowledgeable, preventing
+   * the user from proceeding and relying on actions to proceed.
```
Comment (matthewwalsh0) [inline ui/ducks/confirm-alerts/confirm-alerts.ts:19]: Should we also clarify the default is `false` if not specified?

File: ui/ducks/confirm-alerts/confirm-alerts.ts (hunk 2 above)
Comment (vinistevam) [reply ui/ducks/confirm-alerts/confirm-alerts.ts:21]: Done

File: ui/ducks/confirm-alerts/confirm-alerts.ts (hunk 3 above)
Comment (vinistevam) [reply ui/ducks/confirm-alerts/confirm-alerts.ts:19]: Done

**Review by jpuri — APPROVED** [review]

File: ui/ducks/confirm-alerts/confirm-alerts.ts [hunk 4]
```diff
@@ -24,6 +24,12 @@ export type Alert = {
    */
   key: string;
 
+  /**
+   * Whether the alert is a blocker and un-acknowledgeable, preventing the user
+   * from proceeding and relying on actions to proceed. The default is `false`.
+   */
+  isBlocking?: boolean;
```
Comment (digiwand) [inline ui/ducks/confirm-alerts/confirm-alerts.ts:31]: food for thought: an alternative name would be to use isConfirmRequired or isAcknowledgementRequired. isBlocking seems good to me.

**Review by digiwand — APPROVED** [review]

**Pull request merged**